libs/sync/snapshot_state.go

```diff
@@ -50,8 +51,12 @@ func NewSnapshotState(localFiles []fileset.File) (*SnapshotState, error) {
 		// Compute the remote name the file will have in WSFS
 		remoteName := filepath.ToSlash(f.Relative)
 		isNotebook, _, err := notebook.Detect(f.Absolute)
-		if err != nil {
-			// Ignore this file if we're unable to determine the notebook type.
+
+		// If the error is not a file not found error, skip the file.
+		// If the file is not found, it might have been removed from the local FS.
+		// We still want to sync the removal to the remote FS.
```

Comment:
For my understanding, this happens only if the file is deleted right in between performing the sync and computing a new snapshot, correct?

Reply:
If so, then there is another race if the underlying file actually is a notebook.

The remote name of the file is determined below, depending on the file being a notebook or not. If we include the file as not being a notebook (i.e. the remote path has an extension), while it was a notebook, then we won't be able to delete the remote notebook.


I think the only way to solve this is by explicitly including the remote names in the argument to this constructor.

Reply:
> this happens only if the file is deleted right in between performing the sync and computing a new snapshot, correct?

No, not really, this is when we pull state and don't have local snapshot. `notebook.Detect(f.Absolute)` will return error which meant we skip adding files to snapshot regardless. With the change even if we don't find a file on local FS when it's in deployment state - we add it to snapshot anyway to correctly calculate diff later.

This behaviour actually uncovered by E2E test, so it's covered by tests :)

Reply:
Addressed this by refactoring to have `File.IsNotebook` and storing `isNotebook` in deployment state